OpenRA.Game/Orders/IOrderGenerator.cs

```diff
@@ -16,10 +16,18 @@ namespace OpenRA
 {
 	public interface IOrderGenerator
 	{
-		IEnumerable<Order> Order(World world, CPos cell, int2 worldPixel, MouseInput mi);
+		IEnumerable<Order> Order(World world, CPos cell, int2 worldPixel, MouseInput mi, uint extraData = 0);
```

Comment:
Mixing in a low-level order primitive (`extraData`) with the processed/derived data (everything else) feels like a really bad choice.  Unfortunately I can't dig around in the code to be able to suggest a better approach right now.